### PR TITLE
8014 step duration calendar bug

### PR DESCRIPTION
**Pull request by arisah-ijeoma**

### CODE REVIEW

File: app/models/calendar_event.rb [hunk 1]
```diff
@@ -84,8 +84,10 @@ def self.project_date_filter(range, scope = Project)
   end
 
   def self.project_step_date_filter(range, scope = ProjectStep, project_id)
-    scoped_steps = scope.where("scheduled_start_date <= :last AND "\
-      "COALESCE(actual_end_date, scheduled_start_date + scheduled_duration_days) >= :first",
+    scoped_steps = scope.where(
+      "COALESCE(actual_end_date, scheduled_start_date + scheduled_duration_days) >= :first AND "\
+      "scheduled_start_date <= :last OR "\
+      "scheduled_start_date BETWEEN :first AND :last",
```
Comment (emitche) [inline app/models/calendar_event.rb]: If we just used line 90 and not line 89, would it have the same effect?

Comment (arisah-ijeoma) [reply app/models/calendar_event.rb]: I could test that out
before we moved around the scopes, one couldn't do without the other

**Comment by emitche** [comment]
Functional review: Yes, the bug is fixed with these changes.

**Comment by emitche** [comment]
I merged the most recent develop into this branch.

File: app/models/calendar_event.rb (hunk 1 above)
Comment (emitche) [reply app/models/calendar_event.rb]: Okay. Thanks.

Comment (arisah-ijeoma) [reply app/models/calendar_event.rb]: @emitche, I've tested it and this OR statement is a necessity

**Pull request merged**